lib/active_merchant/billing/gateways/quickbooks.rb

```diff
@@ -170,10 +189,12 @@ def commit(uri, body = {}, method = :post)
         # The QuickBooks API returns HTTP 4xx on failed transactions, which causes a
         # ResponseError raise, so we have to inspect the response and discern between
         # a legitimate HTTP error and an actual gateway transactional error.
+        headers = {}
         response = begin
           case method
           when :post
-            ssl_post(endpoint, post_data(body), headers(:post, endpoint))
+            headers = headers(:post, endpoint)
+            ssl_post(endpoint, post_data(body), headers)
           when :get
             ssl_request(:get, endpoint, nil, headers(:get, endpoint))
```

Comment:
Just want to confirm, we don't need to stash the results of `headers(:get, endpoint)`, only for `headers(:post, endpoint)` above?

Reply:
it doesn't look like we're actually doing any get requests currently, but I think we should probably stash them too so we can send them to the `response_object` if needed